include/Contact.php

```diff
@@ -533,16 +535,27 @@ function get_contact($url, $uid = 0) {
 
 	update_contact_avatar($data["photo"],$uid,$contactid);
 
-	q("UPDATE `contact` SET `addr` = '%s', `alias` = '%s', `name` = '%s', `nick` = '%s',
-		`name-date` = '%s', `uri-date` = '%s' WHERE `id` = %d",
-		dbesc($data["addr"]),
-		dbesc($data["alias"]),
-		dbesc($data["name"]),
-		dbesc($data["nick"]),
-		dbesc(datetime_convert()),
-		dbesc(datetime_convert()),
-		intval($contactid)
-	);
+	$r = q("SELECT `addr`, `alias`, `name`, `nick` FROM `contact`  WHERE `id` = %d", intval($contactid));
+
+	// This condition should always be true
+	if (!dbm::is_result($r))
+		return $contactid;
+
+	// Only update if there had something been changed
+	if (($data["addr"] != $r[0]["addr"]) OR
+		($data["alias"] != $r[0]["alias"]) OR
+		($data["name"] != $r[0]["name"]) OR
+		($data["nick"] != $r[0]["nick"]))
+		q("UPDATE `contact` SET `addr` = '%s', `alias` = '%s', `name` = '%s', `nick` = '%s',
+			`name-date` = '%s', `uri-date` = '%s' WHERE `id` = %d",
+			dbesc($data["addr"]),
+			dbesc($data["alias"]),
+			dbesc($data["name"]),
+			dbesc($data["nick"]),
+			dbesc(datetime_convert()),
+			dbesc(datetime_convert()),
+			intval($contactid)
+		);
```

Comment:
Fair enough. I personally take the following precaution not to confuse condition indentation with the following code indentation:

    if (condition
        on
        multiple
        lines
    ) {
        actions
    }

Instead of this:

    if (condition
        on
        multiple
        lines) {
        actions
    }